### ISSUE TITLE
PERLDB_NOOPT ignored by 'Peephole optimise adjacent pairs of nextstate ops' patch

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#10794]
Migrated from [rt.perl.org#78810](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=78810) (status was 'resolved')

Searchable as RT78810$

**Comment by p5pRT** [comment]
### From @timbunce

#### Created by @timbunce




The effect of the "Peephole optimise adjacent pairs of nextstate ops"
patch \(ac56e7d\) should be disabled when PERLDB\_NOOPT is true\.

<details><summary>Perl Info</summary>

```

Flags:
    category=core
    severity=low

Site configuration information for perl 5.13.6:

Configured by timbo at Sat Oct 30 11:18:14 IST 2010.

Summary of my perl5 (revision 5 version 13 subversion 6) configuration:
  Commit id: cbf9c13fe9f25a21762d4a1f4419e3b0ab947a53
  Platform:
    osname=darwin, osvers=10.4.0, archname=darwin-thread-multi-2level
    uname='darwin timac.local 10.4.0 darwin kernel version 10.4.0: fri apr 23 18:28:53 pdt 2010; root:xnu-1504.7.4~1release_i386 i386 '
    config_args='-des -Doptimize=-g -DEBUGGING=both -Dusethreads -Dusemultiplicity -Dusesitecustomize -Dusedevel -Uversiononly -Accflags=-DDEBUG_LEAKING_SCALARS -Dprefix=/usr/local/perl5-blead-deb-thr'
    hint=recommended, useposix=true, d_sigaction=define
    useithreads=define, usemultiplicity=define
    useperlio=define, d_sfio=undef, uselargefiles=define, usesocks=undef
    use64bitint=undef, use64bitall=undef, uselongdouble=undef
    usemymalloc=n, bincompat5005=undef
  Compiler:
    cc='cc', ccflags ='-fno-common -DPERL_DARWIN -no-cpp-precomp -DDEBUG_LEAKING_SCALARS -DDEBUGGING -fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include -I/opt/local/include',
    optimize='-g',
    cppflags='-no-cpp-precomp -fno-common -DPERL_DARWIN -no-cpp-precomp -DDEBUG_LEAKING_SCALARS -DDEBUGGING -fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include -I/opt/local/include'
    ccversion='', gccversion='4.2.1 (Apple Inc. build 5664)', gccosandvers=''
    intsize=4, longsize=4, ptrsize=4, doublesize=8, byteorder=1234
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=16
    ivtype='long', ivsize=4, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=8
    alignbytes=8, prototype=define
  Linker and Libraries:
    ld='env MACOSX_DEPLOYMENT_TARGET=10.3 cc', ldflags =' -fstack-protector -L/usr/local/lib -L/opt/local/lib'
    libpth=/usr/local/lib /opt/local/lib /usr/lib
    libs=-lgdbm -ldbm -ldl -lm -lutil -lc
    perllibs=-ldl -lm -lutil -lc
    libc=/usr/lib/libc.dylib, so=dylib, useshrplib=false, libperl=libperl.a
    gnulibc_version=''
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=bundle, d_dlsymun=undef, ccdlflags=' '
    cccdlflags=' ', lddlflags=' -bundle -undefined dynamic_lookup -L/usr/local/lib -L/opt/local/lib -fstack-protector'

Locally applied patches:
    


@INC for perl 5.13.6:
    /usr/local/perl5-blead-deb-thr/lib/site_perl/5.13.6/darwin-thread-multi-2level
    /usr/local/perl5-blead-deb-thr/lib/site_perl/5.13.6
    /usr/local/perl5-blead-deb-thr/lib/5.13.6/darwin-thread-multi-2level
    /usr/local/perl5-blead-deb-thr/lib/5.13.6
    /usr/local/perl5-blead-deb-thr/lib/site_perl/5.13.4
    /usr/local/perl5-blead-deb-thr/lib/site_perl/5.11.4
    /usr/local/perl5-blead-deb-thr/lib/site_perl/5.11.3
    /usr/local/perl5-blead-deb-thr/lib/site_perl/5.11.2
    /usr/local/perl5-blead-deb-thr/lib/site_perl
    .


Environment for perl 5.13.6:
    DYLD_LIBRARY_PATH=:/usr/local/pgsql/lib/:/opt/local/lib/mysql5/mysql
    HOME=/Users/timbo
    LANG=en_IE.UTF-8
    LANGUAGE (unset)
    LC_ALL=en_IE.UTF-8
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/usr/local/perl5-blead-deb-thr/bin:/usr/local/pgsql/bin:/Users/timbo/bin:/usr/local/perl512-dev/bin:/Users/timbo/perl6/rakudo/parrot_install/bin:/usr/local/mysql/bin:/usr/local/bin:/opt/local/bin:/opt/local/sbin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/bin:/usr/X11/bin
    PERLCRITIC=/Users/timbo/.setdev/perlcriticrc
    PERLTIDY=/Users/timbo/.setdev/perltidyrc
    PERL_BADLANG (unset)
    SHELL=/bin/bash

```
</details>

**Comment by p5pRT** [comment]
### From @cpansprout

Fixed by bcc76ee\.

**Comment by p5pRT** [comment]
@cpansprout - Status changed from 'new' to 'resolved'
